### PR TITLE
openlineage: integrate provider configuration into Airflow config

### LINKED ISSUE #29670: Integrate OpenLineage provider configuration with general Airflow configuration
### Body

Currently OL-Airflow is configured either with env variables (some things) or `openlineage.yml` file. 
We need to add a way to configure provider using Airflow config the same way as other providers and core Airflow is configured.
At the same time, we need to respect openlineage.yml and env based configs - the structure of new config should be consistent.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by JDarDagran**
This PR adds configuration of the OpenLineage provider with Airflow config. The precedence of Airflow config over existing ones in current `openlineage-airflow` is also described.

Depends on #30708 changes
Closes: #29670

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1159,6 +1159,56 @@ lineage:
...
+    transport:
+      description: |
+        OpenLineage Client transport configuration. It should contain type
+        and additional options per each type.
+        Currently supported types are:
+          * HTTP
+          * Kafka
+          * Console
+      type: string
+      example: '{"type": "http", "url": "http://localhost:5000"}}'
```
Comment (ashb) [inline airflow/config_templates/config.yml]:
```suggestion
      example: '{"type": "http", "url": "http://localhost:5000"}'
```

File: airflow/providers/openlineage/plugins/adapter.py [hunk 2]
```diff
@@ -70,9 +74,34 @@ def __init__(self, client: OpenLineageClient | None = None, secrets_masker: Secr
...
+        # Second, try to get transport config
+        transport = conf.getjson("openlineage", "transport")
+        if not transport:
+            return None
+        elif not isinstance(transport, dict):
+            raise ValueError(f"{transport} is not a dict")
+        return transport
+
+    def _read_yaml_config(self, path: str) -> dict | None:
+        if os.path.exists(path):
```
Comment (ashb) [inline airflow/providers/openlineage/plugins/adapter.py]: If a something is specified but doesn't exist it should probably be a error?

```suggestion
        if path:
```

**Review by ashb — COMMENTED** [review]
Few small Qs, looks good overall though

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (JDarDagran) [reply airflow/config_templates/config.yml]: Fixed.

File: airflow/providers/openlineage/plugins/adapter.py (hunk 2 above)
Comment (JDarDagran) [reply airflow/providers/openlineage/plugins/adapter.py]: Done.

**Review by ashb — APPROVED** [review]

**Pull request merged**